avocado/plugins/runner.py

```diff
@@ -391,23 +387,24 @@ def run_suite(self, job, result, test_suite, variants):
                     if 'methodName' in test_parameters:
                         del test_parameters['methodName']
                     test_factory = (TimeOutSkipTest, test_parameters)
-                    if not self.run_test(job, result, test_factory, queue, summary):
+                    if not self.run_test(job, job.result, test_factory, queue,
+                                         summary):
```

Comment:
For the next round of refactoring, I suggest the `run_test` and `_run_test` methods, as it has `job` and `job.result` as arguments. Also, the docstrings of both methods are not accurate with the method signature.

Reply:
Yep, it is in my queue. ;)